org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js

```diff
@@ -207,6 +207,7 @@ function providerSchemaToPlatformSchema(providerSchema, expand) {
 
         dim.id = 'dim';
         dim.dimmingSetting = providerSchema['attributes'].level;
+        if (dim.dimmingSetting > 100 ) dim.dimmingSetting = 100;
```

Comment:
When I tested AEOTec switch through SmartThings app and checked the result in the ST portal, I did not see any level changes in the log for AOETec switch(the ST app's UI gives the illusion that level is adjusted , but in the backend it was not). The only time I see the dimming level change is between 99% and 255%, and when that happens, the ST always indicate the switch level is at maximum. Therefore, I am just capping the level occuring to the ST documentation (please see the description for this PR).

I am not sure it that weird behavior is due to fact that the switch is not outputting electricity. Maybe you can try to test it with the real hardware. You may check how AEOTec switch really behaves by selecting "AEO switch back wall" in the "live logging" tab of the ST portal.

---
In reply to: [112323616](https://github.com/openT2T/translators/pull/235#discussion_r112323616) [](ancestors = 112323616)